"Improve my website's performance" (#12372)
newrelic/docs-website: src/content/docs/journey-performance/improve-website-performance.mdx

---
title: "Improve your website's performance"
---
import browserCoreWebVitals from 'images/browser_diagram_core-web-vitals.png'

import browserBrowserSummary from 'images/browser_screenshot-full_browser-summary.png'

import browserErrorOverview from 'images/browser_screenshot-full_error-overview.png'

import browserSuggestedServiceLevels from 'images/suggested_service_levels.png'

> A good website—you know it when you surf it, right? It's so fast you never notice the load time. It's available whenever you need it. And, of course, it's error-free. But _measuring_ your site's performance, let alone improving it, is a tricky task.

Reviewer: ```suggestion
A good website — you know it when you surf it, right? It's so fast you never notice the load time. It's available whenever you need it. And, of course, it's error-free. But _measuring_ your site's performance, let alone improving it, is a tricky task.
```

Reviewer: I'm not sure of this, but it looks better spaced out, right?

This tutorial walks you through how to use New Relic to improve your site's experience. You'll start by "instrumenting" your site, which means adding some JavaScript code that measures the actual end-user experience for each and every visit. From there, you'll learn how to understand how your site is performing today, where you can improve, and how to keep an eye on things in future.

## Improve your website's performance

<Steps>

<Step>
### Instrument your site

Before you can improve your site's performance, you need to see what's going on. New Relic monitoring always starts with an **integration**, a piece of code that collects data from a technology you care about and reports it to New Relic. 

> To monitor the front end of your website, the integration you want is our **Browser monitoring agent**. This is a small snippet of JavaScript that you'll include in your HTML `<head>`, very similar to Google Analytics or other scripts. 

Reviewer: ```suggestion
To monitor the front end of your website, the integration you want is our **Browser monitoring agent**. This is a small snippet of JavaScript that you'll include in your HTML `<head>`, similar to adding Google Analytics or other scripts to your site. 
```

Reviewer 2: Changed the structure of this sentence a bit, thanks for the comment!

> To monitor the front end of your website, the integration you want is our **Browser monitoring agent**. This is a small snippet of JavaScript that you'll include in your HTML `<head>`, very similar to Google Analytics or other scripts. 

Reviewer: ```suggestion
To monitor the front end of your website, the integration you want is our **browser monitoring agent**. This is a small snippet of JavaScript that you'll include in your HTML `<head>`, very similar to Google Analytics or other scripts. 
```

> There are two basic ways to install the Browser monitoring agent: 

Reviewer 2: Good catch, thanks!

<Tabs>
    <TabsBar>
        <TabsBarItem id="js-snippet">Copy-paste install</TabsBarItem>
        <TabsBarItem id="apm-agent-deploy">APM agent install</TabsBarItem>
    </TabsBar>

    <TabsPages>
        <TabsPageItem id="js-snippet">
            You can deploy just the agent by copy-pasting the JavaScript snippet into whatever part of your code generates the `<head>` for your webpage.

            <ButtonGroup>
                <ButtonLink
                    role="button"
                    to="https://one.newrelic.com/marketplace?account=3132060&state=39f84c44-d66c-1ec2-12e0-f03316b4a9cd"
                    variant="primary"
                    >
                    Get copy-paste snippet
                </ButtonLink>

                <ButtonLink
                    role="button"
                    to="https://docs.newrelic.com/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/#options"
                    variant="primary"
                    >
                    See install docs
                </ButtonLink>
            </ButtonGroup>
        </TabsPageItem>

        <TabsPageItem id="apm-agent-deploy">

>             If you're monitoring your backend with one of our APM agents, the APM agent can automatically inject the Browser monitoring agent into the appropriate part of your webpage.

Reviewer: ```suggestion
            If you're monitoring your backend with one of our APM agents, the APM agent can automatically inject the browser monitoring agent into the appropriate part of your webpage.
```

Reviewer 2: Good catch, thanks!

                    to="https://one.newrelic.com/nr1-core?  state=fcaeee53-614a-4c29-92ee-694e0b128368"
                />

                <TechTile
                    name="Node.js agent"
                    icon="logo-nodejs"
                    to="https://one.newrelic.com/nr1-core?  state=c872674f-2350-7ec3-125f-a4335255e180"
                />

                <TechTile
                    name="PHP agent"
                    icon="logo-php"
                    to="https://one.newrelic.com/nr1-core?  state=77aab820-3aaf-066b-c3fc-2f3437b403a8"
                />

                <TechTile
                    name="Python agent"
                    icon="logo-python"
                    to="https://one.newrelic.com/nr1-core?  state=1ed82004-2854-1a4a-2a91-69302832ccc0"
                />

                <TechTile
                    name="Ruby agent"
                    icon="logo-ruby"
                    to="https://one.newrelic.com/nr1-core?  state=d69143ab-605c-579b-25bf-cc6e5fee5b80"
                />
            </TechTileGrid>
        </TabsPageItem>
    </TabsPages>
  </Tabs>
</Step>

<Step>
### Review your core web vitals

<SideBySide>
  <Side>

>     Core web vitals are metrics that give you insight into your users' experience with your site. Low scores often indicate a poor experience, which can leave users irritated. New Relic has <InlinePopover type="browser" /> capabilities you can use to track these metrics, and you can see our tutorial on core web vitals for more information.

Reviewer: ```suggestion
    Core web vitals are Google's metrics that give you insight into your users' experience with your site. Low scores often indicate a poor experience, which can deteriorate your relationship with your end users. New Relic has <InlinePopover type="browser" /> capabilities you can use to track these metrics, and you can see our tutorial on core web vitals for more information.
```

Reviewer 2: Better than what I had, thank you!
